From 064807d22a0a2865a930d4cc48b61f917f4bf15f Mon Sep 17 00:00:00 2001
From: Anders Dellien <anders.dellien@arm.com>
Date: Fri, 17 Apr 2020 16:06:03 +0100
Subject: [PATCH 2/2] swap red and blue colors in HDLCD driver

The HDLCD drivers is configured to use BGR888 but Android needs
RGB888 - so we need to swap the red and blue components.

Change-Id: Ib47ddff9280a53757925f6c1b1f373089d038025
Signed-off-by: Anders Dellien<anders.dellien@arm.com>
Upstream-Status: Inappropriate [Color swap needed for TC0 only]
Signed-off-by: Teo Couprie Diaz <teo.coupriediaz@arm.com>
---
 drivers/gpu/drm/arm/hdlcd_crtc.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/arm/hdlcd_crtc.c b/drivers/gpu/drm/arm/hdlcd_crtc.c
index e69d996..a46ba4f 100644
--- a/drivers/gpu/drm/arm/hdlcd_crtc.c
+++ b/drivers/gpu/drm/arm/hdlcd_crtc.c
@@ -110,15 +110,15 @@ static int hdlcd_set_pxl_fmt(struct drm_crtc *crtc)
 	 * pixel is outside the visible frame area or when there is a
 	 * buffer underrun.
 	 */
-	hdlcd_write(hdlcd, HDLCD_REG_RED_SELECT, format->red.offset |
+	hdlcd_write(hdlcd, HDLCD_REG_RED_SELECT, format->blue.offset |
 #ifdef CONFIG_DRM_HDLCD_SHOW_UNDERRUN
 		    0x00ff0000 |	/* show underruns in red */
 #endif
-		    ((format->red.length & 0xf) << 8));
+		    ((format->blue.length & 0xf) << 8));
 	hdlcd_write(hdlcd, HDLCD_REG_GREEN_SELECT, format->green.offset |
 		    ((format->green.length & 0xf) << 8));
-	hdlcd_write(hdlcd, HDLCD_REG_BLUE_SELECT, format->blue.offset |
-		    ((format->blue.length & 0xf) << 8));
+	hdlcd_write(hdlcd, HDLCD_REG_BLUE_SELECT, format->red.offset |
+		    ((format->red.length & 0xf) << 8));
 
 	return 0;
 }
-- 
2.7.4

